components/org.wso2.carbon.identity.api.idpmgt.endpoint/src/main/java/org/wso2/carbon/identity/api/idpmgt/endpoint/EndpointUtils.java

```diff
@@ -0,0 +1,794 @@
+
+        FederatedAuthenticatorConfigDTO federatedAuthenticatorConfigDTO = null;
+        if (federatedAuthenticatorConfig != null) {
+            federatedAuthenticatorConfigDTO = new FederatedAuthenticatorConfigDTO();
+            federatedAuthenticatorConfigDTO.setDisplayName(federatedAuthenticatorConfig.getDisplayName());
+            federatedAuthenticatorConfigDTO.setEnabled(federatedAuthenticatorConfig.isEnabled());
+            federatedAuthenticatorConfigDTO.setName(federatedAuthenticatorConfig.getName());
+
+            if (ArrayUtils.isNotEmpty(federatedAuthenticatorConfig.getProperties())) {
+                List<PropertyDTO> propertyList = getPropertyListDTO(Arrays.asList(federatedAuthenticatorConfig.getProperties()));
+                federatedAuthenticatorConfigDTO.setPropertyList(propertyList);
+            }
+        }
+        return federatedAuthenticatorConfigDTO;
+    }
+
+    private static List<PropertyDTO> getPropertyListDTO(List<Property> properties) {
+
+        List<PropertyDTO> propertyList = new ArrayList<>();
+        if (properties != null) {
+            properties.forEach(property -> {
+                PropertyDTO propertyDTO = new PropertyDTO();
+                propertyDTO.setValue(property.getValue());
+                propertyDTO.setAdvanced(property.isAdvanced());
+                propertyDTO.setConfidential(property.isConfidential());
+                propertyDTO.setDefaultValue(property.getDefaultValue());
+                propertyDTO.setDisplayName(property.getDescription());
+                propertyDTO.setDescription(property.getDisplayName());
+                propertyDTO.setDisplayOrder(property.getDisplayOrder());
+                propertyDTO.setName(property.getName());
+                propertyDTO.setRequired(property.isRequired());
+                propertyDTO.setType(property.getType());
+                propertyDTO.setValue(property.getValue());
+                propertyList.add(propertyDTO);
+            });
+        }
+
+        return propertyList;
+    }
+
+    private static List<Property> getPropertyList(List<PropertyDTO> propertyDTOS) {
+
+        List<Property> propertyList = new ArrayList<>();
+        if (propertyDTOS != null) {
+            propertyDTOS.forEach(propertyDTO -> {
+                Property property = new Property();
+
+                property.setValue(propertyDTO.getValue());
+                if (propertyDTO.getAdvanced() != null) {
+                    property.setAdvanced(propertyDTO.getAdvanced());
+                }
+                if (propertyDTO.getConfidential() != null) {
+                    property.setConfidential(propertyDTO.getConfidential());
+                }
+                property.setDefaultValue(propertyDTO.getDefaultValue());
+                property.setDisplayName(propertyDTO.getDescription());
+                property.setDescription(propertyDTO.getDisplayName());
+                if (propertyDTO.getDisplayOrder() != null) {
```

Comment:
You do check null for certain values and don't for some other. Any specific reason ? Cant we just set them without checking for null ?